packages/flutter/lib/src/widgets/image.dart

```diff
@@ -1066,9 +1065,12 @@ class _ImageState extends State<Image> with WidgetsBindingObserver {
   @override
   void didChangeDependencies() {
     _updateInvertColors();
-    _resolveImage();
 
+    // Resolve and decode the image only if ticker mode is not disabled.
+    // Thus, if ticker mode is disabled in the whole lifecycle of the image,
+    // it does not need to get a stream.
     if (TickerMode.of(context)) {
+      _resolveImage();
```

Comment:
Actually, I am not convinced that this is the right fix. This would actually delay the image loading until you're in a context with enabled tickerprovider, which could result in flickering.

Reply:
First of all, I want to sincerely thank you for taking the time to review my PR, despite my limited skills and English proficiency.

**As I understand it**, `_resolveImage()` is responsible for fetching the image information, while the actual rendering on the screen is handled by `_listenToStream()`. Before my PR, originally, if an `Image` widget is a child of a widget with `TickerMode.enabled` set to `false`, the image won't be displayed. This is likely because `_listenToStream()` is not executed. As explained in #13106, when `TickerMode` is disabled, there is no need to render the image, so resolving the image should not be necessary either. It seems natural to first resolve the image only when rendering is required, i.e., when `TickerMode` is no longer disabled. While I'm not entirely sure what scenario "flickering" refers to, even if there is a delay in loading the image, I believe it is appropriate that resolving the image should also be switched when toggling `TickerMode`.

If minimizing the delay in image resolving is a priority, it might be necessary to always resolve the image while asynchronously removing the error handler, as mentioned in the PR. However, since `reportError` is connected via a `then()` chain in the `MultiFrameImageStreamCompleter` constructor, it seems difficult to `await` it. If you have any good ideas, I would appreciate your input.

Reply:
I am trying to understand the fix. wouldn't you still run into crashes when
the following sequnce

frame 1: ticker enabled and _resolveImage is called (which will throw in frame 3)
frame 2: ticker disabled, the remove stream listener
frame 3: throw and unhandled crashes

Reply:
Do you have any reproducible code for the sequence you described?

Reply:
Are you referring to a scenario where the ticker is disabled before `_resolveImage` called in frame 1 has completed?

Reply:
@enihsgnir were you able to figure out if the scenario outlined by @chunhtai would cause any issues?